### ISSUE TITLE
path in the breakpoint expression no working

### ISSUE THREAD

**Issue by lies2020** [x64dbg/x64dbg#3256]
### Operating System

Windows 10

### x64dbg Version

Oct 29 2023

### Describe the issue

when you edit the breakpoint，the back slash("\\") in string  is converted into an escape character("\\\\")
edit the breakpoint again,  escape character("\\\\")  is converted into "\\\\\\\\"

### Steps to reproduce

1. set breakpoints at <kernel32.dll.CreateFileW>, edit breakpoint, break condition: streq(utf16(arg(0)),"D:\FSViewer\FSIV02.fslang"}
2. save, break condition become: streq(utf16(arg(0)),"D:\\\\FSViewer\\\\FSIV02.fslang"}
3. Run debug, when reaching breakpoint, can not break and log show "Error when evaluating break condition."
4. Edit breakpoint again, do nothing and save, break condition become: streq(utf16(arg(0)),"D:\\\\\\\\FSViewer\\\\\\\\FSIV02.fslang"}

**Comment by shocoman** [comment]
Does anybody know a reason for escaping backslashes [here](https://github.com/x64dbg/x64dbg/blob/b0392eda1c8d96127df1e32c46ab3481e1c26483/src/gui/Src/Gui/EditBreakpointDialog.cpp#L68)? Except that they affect the quote character (`"`).
It seems even before [my last PR ](https://github.com/x64dbg/x64dbg/pull/3250), which certainly exacerbated this problem, there was an issue when the Edit Breakpoint dialog would turn a normal expression like `streq(utf8(AABBCCDD),"A\B\C")` into an invalid one like `streq(utf8(AABBCCDD),"A\\B\\C")`

**Comment by mrexodia** [comment]
Previously the \\ just turned back into a \ after processing I think. It was an oversight that the quoting isn’t adjusted accordingly

**Comment by shocoman** [comment]
> Previously the \ just turned back into a \ after processing I think. It was an oversight that the quoting isn’t adjusted accordingly

It looks like Markdown ate some of your backslashes.

If currently those double backslashes stay unchanged in the interpreter (or wherever they go), would it be okay if this dialog simply did not escape `\`s? 
I've also noticed that it's impossible to create a quoted string that would end with just `\"` (as two separate characters), because `\"` turns into `"`, `\\"` into `\\`, and `\\\"` into `\\"`.

**Comment by mrexodia** [comment]
Yeah there was a change recently in the command parser that likely caused this to become so problematic: https://github.com/x64dbg/x64dbg/pull/3217. The parser is meant for commands where it needs to kinda guess the 'intent' of the user. For example you do not want to have a user type `InitDebug "C:\\my\\path\\blah.exe"`. As a result shortcuts are taken with the quoting, but these issues are compounding with the complex expressions that are possible for the breakpoints...

I'm not sure what the best solution is. Probably it would be some for of round-trip thing where a quoting function is exposed by the debugger. Then we can test if this actually works for all strings...
